geode-core/src/main/java/org/apache/geode/internal/cache/AbstractUpdateOperation.java

```diff
@@ -179,8 +186,15 @@ public static boolean doPutOrCreate(LocalRegion rgn, EntryEventImpl ev, long las
             br.getPartitionedRegion().getPrStats().startApplyReplication();
           }
           try {
-            if (rgn.basicUpdate(ev, false/* ifNew */, true/* ifOld */, lastMod,
-                overwriteDestroyed)) {
+            boolean secondAttempt = false;
+            try {
+              secondAttempt = rgn.basicUpdate(ev, false/* ifNew */, true/* ifOld */, lastMod,
```

Comment:
I think this part's logic is incorrect. And the variable secondAttempt should be changed to a better name or omitted.

The rgn.basicUpdate can only have 3 possible results:
1) return true: which means the . update succeeded. Then we should return.
2) return false but not due to CME. It should be caused by the existing entry was removed by another thread. So update with ifOld==true failed. In that case, we should retry with basicUpdate(ifNew==false, ifOld==false, throwConcurrentModificaiton==false), i.e. do update and ignore the possible CME.
3) return false due to CME. It should retry with basicUpdate(ifNew==false, ifOld==false, shouldNotify==false, throwConcurrentModificaiton==false)

You can see in both case 2) and 3), we need to retry basicUpdate(ifNew==false, ifOld==false, , throwConcurrentModificaiton==false). The only difference is: if there's CME, the shouldNotify==false.

From above analysis, variable secondAttempt can be omitted.